utils/cpuload/cpuload.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cpuload
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/golang/glog"
+)
+
+type CpuLoadReader struct {
+	familyId uint16
+	conn     *Connection
+}
+
+func NewCpuLoadReader() (*CpuLoadReader, error) {
+	conn, err := NewConnection()
+	if err != nil {
+		fmt.Printf("failed to create new connection")
+		return nil, err
+	}
+
+	id, err := getFamilyId(conn)
+	if err != nil {
+		fmt.Printf("failed to get family id")
+		return nil, err
+	}
+	glog.V(2).Infof("Family id for taskstats: %d", id)
+	return &CpuLoadReader{
+		familyId: id,
+		conn:     conn,
+	}, nil
+}
+
+func (self *CpuLoadReader) Close() {
+	if self.conn != nil {
+		self.conn.Close()
+	}
+}
+
+// This mirrors kernel internal structure.
+type LoadStats struct {
+	NrSleeping       uint64 // Number of tasks sleeping
+	NrRunning        uint64 // Number of tasks running
+	NrStopped        uint64 // Number of tasks in stopped state
+	NrUinterruptible uint64 // Number of tasks in uninterruptible state
+	NrIoWait         uint64 // Number of tasks waiting on IO
+}
+
+// Returns instantaneous number of running tasks in a group.
+// Caller can use historical data to calculate cpu load.
+// path is a cgroup path under cpu hierarchy eg. "/sys/fs/cgroup/cpu/cadvisor"
```

Comment:
"path is an absolute filesystem path for a container under the CPU cgroup hierarchy"?